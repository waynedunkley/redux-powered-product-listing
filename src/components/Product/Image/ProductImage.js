import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import imageStyles from "./ProductImage.style"

const StyledImage = styled.div`
  ${imageStyles};
`

const propTypes = {
  url: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
}

const ProductImage = ({ url, alt }) => (
  <StyledImage>
    <img src="" alt="" />
  </StyledImage>
)

ProductImage.propTypes = propTypes

export default ProductImage
