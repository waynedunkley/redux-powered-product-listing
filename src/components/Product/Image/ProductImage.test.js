import React from "react"
import { shallow } from "enzyme"
import ProductImage from "./ProductImage"

it("ProductDescription renders description", () => {
  const wrapper = shallow(<ProductImage url="/path/to/img" alt="alt text" />)
  expect(wrapper).toMatchSnapshot()
})
