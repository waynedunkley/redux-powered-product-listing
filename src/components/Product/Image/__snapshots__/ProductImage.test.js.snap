// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDescription renders description 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductImage
    alt="alt text"
    url="/path/to/img"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <img
        alt="alt text"
        src="/path/to/img"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "alt text",
        "src": "/path/to/img",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <img
          alt="alt text"
          src="/path/to/img"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "alt text",
          "src": "/path/to/img",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
